README.md

```diff
@@ -49,10 +49,13 @@ The Make and CMake build systems create three libraries: libmbedcrypto, libmbedx
 
 You need the following tools to build the library with the provided makefiles:
 
-* GNU Make or a build tool that CMake supports.
+* GNU Make 3.82 or a build tool that CMake supports.
 * A C99 toolchain (compiler, linker, archiver). We actively test with GCC 5.4, Clang 3.8, IAR8 and Visual Studio 2013. More recent versions should work. Slightly older versions may work.
-* Python 3 to generate the test code, and to generate sample programs in the development branch.
+* Python 3.6 to generate the test code, and to generate sample programs in the development branch.
 * Perl to run the tests, and to generate some source files in the development branch.
+* CMake 3.10.2 or later (if using CMake).
+* Microsoft Visual Studio 2013 or later (if using Visual Studio).
+* Doxygen 1.8.11 or later (slightly older versions should work).
```

Comment:
I see how it looks tidy to group tool versions in one place, but I don't think this presents information in a useful way. There's what you need to build the library, and there's what you need for fancier things like building the documentation. You don't need Doxygen to “build the library with the provided makefiles” if you don't care about rendering the documentation.

I admit that I previously already made this a little messy by listing tools that are only needed on the development branch, but it is true that Perl is needed to just build the library on the development branch.

I would prefer to keep the Doxygen version listed separately as it was before, but I'd settle for adding something like “To render the documentation” to this bullet point.

Reply:
I think it's better to be consistent and put all this info in one place. Arguably this section could be split into "essential tools" and "optional extras", but this might be overkill? I've added a note as suggested so it's more obviously not a requirement for a basic build without docs.